app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -82,33 +109,76 @@ class HomeFragment : Fragment() {
 
         val inputAsString = context.assets.open(HOME_TILES_JSON_PATH).bufferedReader().use { it.readText() }
         val jsonArray = JSONObject(inputAsString).getJSONArray(HOME_TILES_JSON_KEY)
-        for (i in 0..(jsonArray.length() - 1)) {
-            val jsonObject = jsonArray.getJSONObject(i)
-            homeTiles.add(BundledHomeTile.fromJSONObject(jsonObject))
-        }
-
-        adapter = HomeTileAdapter(onUrlEnteredListener, homeTiles)
+        unpackJsonHomeTiles(homeTiles, jsonArray)
+        val hta = HomeTileAdapter(onUrlEnteredListener, homeTiles)
+        adapter = hta
+        homeTileAdapter = hta
         layoutManager = GridLayoutManager(context, COL_COUNT)
         setHasFixedSize(true)
     }
 
+    private fun unpackJsonHomeTiles(homeTiles: MutableList<HomeTile>, homeTilesJsonArray: JSONArray) {
+        val homeTilesPrefs = context.getSharedPreferences(HOME_TILES_PREFS, MODE_PRIVATE)
+        val blacklist = homeTilesPrefs.getStringSet(bundled_blacklist, null)
+        for (i in 0..(homeTilesJsonArray.length() - 1)) {
+            val jsonObject = homeTilesJsonArray.getJSONObject(i)
+            val id = jsonObject.getString("id")
+            if (blacklist == null || !blacklist.contains(id)) {
+                homeTiles.add(BundledHomeTile.fromJSONObject(jsonObject))
+            }
+        }
+    }
+
     private fun initUrlInputView() = with (urlInputView) {
         setOnCommitListener {
             onUrlEnteredListener.onTextInputUrlEntered(text.toString(), urlInputView.lastAutocompleteResult, UrlTextInputLocation.HOME)
         }
         setOnFilterListener { searchText, view -> urlAutoCompleteFilter.onFilter(searchText, view) }
     }
 
+    fun addBundledTileToBlacklist(id: String) {
```

Comment:
This should be moved to the BundledHomeTilesManager, with the exception of `removeItemAtPosition`